src/plugins/discover/public/application/view_alert/view_alert_route.tsx

```diff
@@ -19,11 +19,34 @@ import { getAlertUtils, QueryParams, SearchThresholdAlertParams } from './view_a
 
 type NonNullableEntry<T> = { [K in keyof T]: NonNullable<T[keyof T]> };
 
-const getCurrentChecksum = (params: SearchThresholdAlertParams) =>
-  sha256.create().update(JSON.stringify(params)).hex();
+const getDataViewParamsChecksum = (dataViewSpec: DataViewSpec) => {
+  const { title, timeFieldName, sourceFilters, runtimeFieldMap } = dataViewSpec;
```

Comment:
I don't think changing the source filters changes the output.

Reply:
Yes, but it won't alter the output of the alert.

Reply:
Yes, but here we are deciding whether to show `Data View has changed` toast message. When he/she opens the alert results and data view has changed in the meantime, we detect it by comparing checksum.

Reply:
@dimaanj 

>When he/she opens the alert results and data view has changed in the meantime, we detect it by comparing checksum.

Yes, I understand that, but maybe there's something else I'm missing. Can you describe the steps a user might take where a change to source filters deserves this toast?

Reply:
Sure, I've updated description